porch/engine/pkg/engine/options.go

```diff
@@ -46,13 +42,29 @@ func WithCache(cache *cache.Cache) EngineOption {
 	})
 }
 
+func WithBuiltinFunctionRuntime() EngineOption {
+	return EngineOptionFunc(func(engine *cadEngine) error {
+		runtime := newBuiltinRuntime()
+		if engine.runtime == nil {
+			engine.runtime = runtime
+		} else {
+			engine.runtime = fn.NewMultiRuntime([]fn.FunctionRuntime{engine.runtime, runtime})
+		}
+		return nil
```

Comment:
It will be 
```
MultiRuntime(
  MultiRuntime(
    first-runtime,
    second-runtime,
  ),
  third-runtime,
)
```
instead of
```
MultiRuntime(
  first,
  second,
  third,
)
```

The correctness is NOT a problem here, since `first-runtime` will be tried first even though it's nested.
I can tweak it if you don't like the nested appraoch.

Reply:
I understand that the code in the PR will look for functions in correct order. my observation is that it defeats the purpose of MultiRuntime; we only use it as TwoRuntime, that's all.

Reply:
Updated